File: konserkita_api/app/Models/Organizer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Organizer extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'company_name',
        'description',
        'logo',
        'website',
        'slug',
        'public_name',
        'cover_image',
        'instagram_url',
        'description_en',
        'rating_average',
        'total_reviews',
        'total_events',
        'total_followers',
        'verification_badge',
        'status',
    ];

    protected $casts = [
        'verification_badge' => 'boolean',
        'rating_average' => 'decimal:2',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function events(): HasMany
    {
        return $this->hasMany(Event::class);
    }

    public function followers(): BelongsToMany
    {
        return $this->belongsToMany(User::class, 'organizer_followers', 'organizer_id', 'user_id')->withTimestamps();
    }

    public function reviews(): HasMany
    {
        return $this->hasMany(OrganizerReview::class);
    }

    protected function logo(): \Illuminate\Database\Eloquent\Casts\Attribute
    {
        return \Illuminate\Database\Eloquent\Casts\Attribute::make(
            get: fn ($value) => $value ? $value : url('default_img.png'),
        );
    }

    protected function coverImage(): \Illuminate\Database\Eloquent\Casts\Attribute
    {
        return \Illuminate\Database\Eloquent\Casts\Attribute::make(
            get: fn ($value) => $value ? $value : url('default_img.png'),
        );
    }
}

File: konserkita_mobile/lib/models/organizer.dart
class Organizer {
  final int id;
  final String companyName;
  final String? publicName;
  final String? slug;
  final String? logo;
  final String? coverImage;
  final bool verificationBadge;
  final String status;
  final double ratingAverage;
  final int totalReviews;

  Organizer({
    required this.id,
    required this.companyName,
    this.publicName,
    this.slug,
    this.logo,
    this.coverImage,
    this.verificationBadge = false,
    this.status = 'pending',
    this.ratingAverage = 0.0,
    this.totalReviews = 0,
  });

  factory Organizer.fromJson(Map<String, dynamic> json) {
    return Organizer(
      id: json['id'],
      companyName: json['company_name'],
      publicName: json['public_name'],
      slug: json['slug'],
      logo: json['logo'],
      coverImage: json['cover_image'],
      verificationBadge: json['verification_badge'] == 1 || json['verification_badge'] == true,
      status: json['status'] ?? 'pending',
      ratingAverage: (json['rating_average'] ?? 0).toDouble(),
      totalReviews: json['total_reviews'] ?? 0,
    );
  }
}
